tests/components/pi_hole/test_init.py

```diff
@@ -88,37 +109,56 @@ async def test_setup_name_config(hass):
     )
 
 
-async def test_disable_service_call(hass):
-    """Test disable service call with no Pi-hole named."""
+async def test_switch(hass, caplog):
+    """Test Pi-hole switch."""
     mocked_hole = _create_mocked_hole()
     with _patch_config_flow_hole(mocked_hole), _patch_init_hole(mocked_hole):
         assert await async_setup_component(
             hass,
             pi_hole.DOMAIN,
-            {
-                pi_hole.DOMAIN: [
-                    {"host": "pi.hole1", "api_key": "1"},
-                    {"host": "pi.hole2", "name": "Custom", "api_key": "2"},
-                ]
-            },
+            {pi_hole.DOMAIN: [{"host": "pi.hole1", "api_key": "1"}]},
         )
 
         await hass.async_block_till_done()
 
         await hass.services.async_call(
-            pi_hole.DOMAIN,
-            pi_hole.SERVICE_DISABLE,
-            {pi_hole.SERVICE_DISABLE_ATTR_DURATION: "00:00:01"},
+            switch.DOMAIN,
+            switch.SERVICE_TURN_ON,
+            {"entity_id": SWITCH_ENTITY_ID},
             blocking=True,
         )
+        mocked_hole.enable.assert_called_once()
 
-        await hass.async_block_till_done()
+        await hass.services.async_call(
+            switch.DOMAIN,
+            switch.SERVICE_TURN_OFF,
+            {"entity_id": SWITCH_ENTITY_ID},
+            blocking=True,
+        )
+        mocked_hole.disable.assert_called_once_with(True)
 
-        assert mocked_hole.disable.call_count == 2
+        # Failed calls
+        type(mocked_hole).enable = AsyncMock(side_effect=HoleError("Error1"))
+        await hass.services.async_call(
+            switch.DOMAIN,
+            switch.SERVICE_TURN_ON,
+            {"entity_id": SWITCH_ENTITY_ID},
+            blocking=True,
+        )
+        type(mocked_hole).disable = AsyncMock(side_effect=HoleError("Error2"))
+        await hass.services.async_call(
+            switch.DOMAIN,
+            switch.SERVICE_TURN_OFF,
+            {"entity_id": SWITCH_ENTITY_ID},
+            blocking=True,
+        )
+        errors = [x for x in caplog.records if x.levelno == logging.ERROR]
+        assert errors[-2].message == "Unable to enable Pi-hole: Error1"
+        assert errors[-1].message == "Unable to disable Pi-hole: Error2"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)